gooddata-sdk/gooddata_sdk/catalog/workspace/service.py

```diff
@@ -24,6 +26,59 @@
 ValidObjectsInputType = Union[ValidObjectTypes, List[ValidObjectTypes], ExecutionDefinition]
 
 
+class CatalogWorkspaceService:
+    def __init__(self, api_client: GoodDataApiClient) -> None:
+        self._client = api_client
+        self._entities_api = metadata_apis.EntitiesApi(api_client.metadata_client)
+
+    def list_workspaces(self) -> List[CatalogWorkspace]:
+        get_workspaces = functools.partial(
+            self._entities_api.get_all_entities_workspaces,
+            include=["workspaces"],
+            _check_return_type=False,
+        )
+        workspaces = load_all_entities(get_workspaces)
+        return [CatalogWorkspace.from_api(w) for w in workspaces.data]
+
+    def create_or_update(self, workspace: CatalogWorkspace) -> None:
```

Comment:
I do not see create scenario implemented, only update and error scenarios.

Reply:
Line 58 implements create workspace scenario (if workspace is not found -> NotFoundException is raised and create is executed).